### DISCUSSION TITLE
Post-install configuration docs

### DISCUSSION THREAD

**Discussion by tomleo** [LycheeOrg/Lychee#2300] [Ideas]
### Enhancement

The docs seem very focused around getting a Lychee LAMP stack up & running, and generally seems like the audience of those docs are developers familiar with Laravel / PHP / RDBMS de jour (mysql or postgresql) / HTTP Server du jour (nginx or apache)

Coming from a python background, it wasn't immediatley obvious what `php artisan` was / does, or available commands.

I have a number of family albums I'd like to move from Google to a self-hosted service. I had no problem getting Lychee up and going on my home server, but I not sure how to get my family photos into the service. I'm assuming using rsync to the nginx folder would be wrong as metadata needs to be generated.

tl;dr would be helpful if there was a little more documentation around `lychee:sync` or other CMD commands to migrate photos from a computer to a lychee instance (ideally over SSH or local folder on the server if possible)

Apologies if I totally missed some obvious documentation on this

**Comment by d7415** [comment]
> would be helpful if there was a little more documentation around `lychee:sync` or other CMD commands to migrate photos from a computer to a lychee instance (ideally over SSH or local folder on the server if possible)

Yeah, we don't really have any documentation for migration, though that would be a good addition (https://github.com/LycheeOrg/LycheeOrg.github.io if you feel like contributing one!).

We have a [FAQ](https://lycheeorg.github.io/docs/faq_general.html#can-i-set-up-lychee-to-watch-a-folder-for-new-images-and-automatically-add-them-to-albums) about sync but it's not super obvious for your use case. Essentially put the photos on the server and use the sync command to import them, then you can optionally remove the originals (unless you use `--import_via_symlink`).


```sh
php artisan lychee:sync --help
Description:
  Sync a directory to lychee

Usage:
  lychee:sync [options] [--] <dir>...

Arguments:
  dir                                            directory to sync

Options:
      --album_id[=ALBUM_ID]                      Album ID to import to
      --owner_id[=OWNER_ID]                      Owner ID of imported photos [default: "1"]
      --resync_metadata                          Re-sync metadata of existing files
      --delete_imported[=DELETE_IMPORTED]        Delete the original files [default: "0"]
      --import_via_symlink[=IMPORT_VIA_SYMLINK]  Imports photos from via a symlink instead of copying the files [default: "0"]
      --skip_duplicates[=SKIP_DUPLICATES]        Skip photos and albums if they already exist in the gallery [default: "0"]
  -h, --help                                     Display help for the given command. When no command is given display help for the list command
  -q, --quiet                                    Do not output any message
  -V, --version                                  Display this application version
      --ansi|--no-ansi                           Force (or disable --no-ansi) ANSI output
  -n, --no-interaction                           Do not ask any interactive question
      --env[=ENV]                                The environment the command should run under
  -v|vv|vvv, --verbose                           Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug```

**Comment by audioscavenger** [comment]
after a recent unexpected upgrade from v4 to v5, my admin user name changed to a blurb $x5dsklfjsdlkfl568ydf6sdf6" and i had to recreate it via artisan.
* Original command `php artisan xyz` did not workj anymore. I had to type the full path like so: `php /app/www/artisan lychee:create_user xyz`
* no mention of that in the docs anywhere, lychee.org, git, google, nothing anywhere
* i spent half an hour pulling my hairs on it but succeeded, other may just give up

What was the issue? Was it me because I did not check the release notes? Was it because of my bad practice to have *latest* tag in the compose?
How can we improve? how do we submit updated documentation?

**Reply by d7415** [reply]
Firstly, this is off topic for this discussion, but I'll answer here anyway.

> * Original command `php artisan xyz` did not workj anymore. I had to type the full path like so: `php /app/www/artisan lychee:create_user xyz`

Were you in the correct directory?

> How can we improve? how do we submit updated documentation?

As referenced above, https://github.com/LycheeOrg/LycheeOrg.github.io

**Reply by audioscavenger** [reply]
So i just submit PR to that one? easy enough then, thanks. I promise you won't regret to have answered an off-topic question!

And to close it, no I was not in the correct directory since I use the compose docker lychee from Linuxserver.io, therefore do not pass full paths or cd commands to docker. One passes docker commands to call artisan, and doing so, never expects to have to do a cd before sending the command. It worked in v4, not in v5 anymore, is all I said. Maybe an environment issue from Linuxserver container, I won't bother to know tbh.

**Reply by ildyria** [reply]
> upgrade from v4 to v5, [...] my admin user name changed to a blurb $x5dsklfjsdlkfl568ydf6sdf6"

This is not a v5 specific chance. This has been live since v4.6.0.
If you had migrated to version 4.6.0 and logged in, the chance would have been taken place invisibly as we have legacy process in place.
https://lycheeorg.github.io/docs/releases.html#v460
That change is already 1.5 year old.
